src/pug/guides/code/python/interactive/5.py

```diff
@@ -0,0 +1,86 @@
+import asyncio
+import pyautogui
+import math
+from requests import Session
+from beam_interactive import start
+from beam_interactive import proto
+from random import random
+
+path = "https://beam.pro/api/v1"
+auth = {
+    "username": "connor",
+    "password": "password"
+}
+
+def login(session, username, password):
+    """Log into the Beam servers via the API."""
+    auth = dict(username=username, password=password)
+    return session.post(path + "/users/login", auth).json()
+
+def get_tetris(session, channel):
+    """Retrieve interactive connection information."""
+    return session.get(path + "/tetris/{id}/robot".format(id=channel)).json()
+
+def on_error(error, conn):
+    """This is called when we get an Error packet. It contains
+    a single attribute, 'message'.
+    """
+    print('Oh no, there was an error!')
+    print(error.message)
+
+def on_report(report, conn):
+    # Reports from beam will end up here
+    currentMouseX, currentMouseY = pyautogui.position()
+    if report.joystick[0]:
+        x = report.joystick[0].coordMean.x;
+        y = report.joystick[0].coordMean.y;
+        # If we have x and y coordinates from the joystick
+        if not math.isnan(x) and not math.isnan(y):
+            pyautogui.moveTo(currentMouseX + 300 * x, currentMouseY + 300 * y)
+
+loop = asyncio.get_event_loop()
+
+@asyncio.coroutine
+def connect():
+    # Initialize session, authenticate to Beam servers, and retrieve Tetris
+    # address and key.
+    session = Session()
+    # When we login we can retrieve the channel id from the response.
+    channel_id = login(session, **auth)['channel']['id']
```

Comment:
I would probably manually pass in the keywords here instead of using `**`. It makes it clear to someone just scanning the code what's going on without having to reference back up